controllers/secretagentconfiguration_controller.go

```diff
@@ -79,51 +79,22 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 	}
 	log.V(1).Info("** Reconcile loop start **")
 
-	// TODO remove this and move setup of secret manager to a container
-	// add conatiner to GenKeyConfig
-
-	//	secretMgr := secretmanager.NewSecretManager(rclient, instance.AppConfig)
-
-	if instance.Spec.AppConfig.SecretsManager != v1alpha1.SecretsManagerNone &&
-		instance.Spec.AppConfig.CredentialsSecretName != "" {
-		var dir string
-		var cloudCredNS string
+	rClient := reconciler.Client
 
-		if len(reconciler.CloudSecretsNamespace) > 0 {
-			cloudCredNS = reconciler.CloudSecretsNamespace
-		} else {
-			cloudCredNS = instance.Namespace
-		}
+	cloudCredNS := reconciler.CloudSecretsNamespace
 
-		// load credentials secret
-		secObject, err := k8ssecrets.LoadSecret(reconciler.Client,
-			instance.Spec.AppConfig.CredentialsSecretName, cloudCredNS)
-		if err != nil {
-			log.Error(err, "error loading cloud credentials secret from the Kubernetes API",
-				"secret_name", instance.Spec.AppConfig.CredentialsSecretName,
-				"cloud_secret_namespace", cloudCredNS)
-			return ctrl.Result{}, err
-		}
-		// Create temporary dir for gcp credentials if needed
-		if instance.Spec.AppConfig.SecretsManager == v1alpha1.SecretsManagerGCP {
-			dir, err = ioutil.TempDir("", "cloud_credentials-*")
-			if err != nil {
-				log.Error(err, "couldn't create a temporary credentials dir")
-				return ctrl.Result{}, err
-			}
-			// clean up after ourselves
-			defer os.RemoveAll(dir)
-		}
+	// Create new Secret Manager object
+	if ctx, sm, err = secretsmanager.NewSecretManager(ctx, &instance, cloudCredNS, rClient); err != nil {
+		return ctrl.Result{}, err
+	}
 
-		// load cloud credentials to envs and/or files
-		if err := manageCloudCredentials(instance.Spec.AppConfig.SecretsManager, secObject, dir); err != nil {
-			log.Error(err, "error loading cloud credentials from secret provided",
-				"secret_name", instance.Spec.AppConfig.CredentialsSecretName,
-				"cloud_secret_namespace", cloudCredNS)
-			return ctrl.Result{}, err
+	// Close client after reconcile loop finishes
+	defer func() {
+		if err := (*sm).CloseClient(); err != nil {
+			log.Error(err, "Problem closing client")
```

Comment:
why not `defer sm.CoseClient()`

Reply:
Because this was the only way I can capture the error. You can only defer 1 thing

Reply:
can't you just log the error in the closeclient method? You're not doing anything here with it, right?

Reply:
Yes you're right. I'll change that.